docs: update apple silicon doc with safetensors `0.8.0` benefits (#46744)
huggingface/transformers: docs/source/en/perf_train_special.md

MPS requires the entire model to fit in unified memory, so `device_map="auto"` can't offload layers to the CPU like CUDA. In this case, try using a smaller model.

> Since safetensors `0.8.0`, models can be loaded directly to MPS with `device_map="mps"` or `"auto"` using zero-copy loading. Weights are mapped into Metal buffers without an intermediate copy, which roughly halves the memory footprint during loading and makes it about 5-6x faster. This requires PyTorch 2.9 or later; older versions fall back to the standard load path.

Reviewer: ```suggestion
Loading weights to MPS is faster and uses less memory with safetensors `0.8.0` and PyTorch 2.9 or later. When `device_map="mps"` or `"auto"`, weights are mapped into Metal buffers without an intermediate copy, which roughly halves the memory footprint during loading and makes it about 5-6x faster. On older PyTorch versions, loading will fall back to the standard load path.
```
